client/landing/gutenboarding/path.ts

```diff
@@ -96,3 +104,32 @@ export function useCurrentStep() {
 export function useNewQueryParam() {
 	return new URLSearchParams( useLocation().search ).has( 'new' );
 }
+
+export function useIsAnchorFm(): boolean {
+	const podcastId = useAnchorFmPodcastId();
+	return podcastId !== null && podcastId !== '';
+}
+
+// Returns the anchor podcast id. First looks in "location state",
+// provided by react-router-dom, if not available there, checks the query string.
+export function useAnchorFmPodcastId(): string | null {
+	const { state: locationState, search } = useLocation< GutenLocationStateType >();
+	const sanitizePodcastId = ( id: string ) => id.replace( /[^a-zA-Z0-9]/g, '' );
```

Comment:
Surprised the new URLSearchParams API doesn't already sanitize params for you. Also surprised that we don't have a helper function to do that?

Reply:
> I don't think we need to sanitize here.

@alshakero Thanks! Would you mind clarifying why we don't need to sanitize here?